Edit updates with the update title

cverna opened the pull request:
This commit makes possible to edit an update using the update title
Fixes #1409

Signed-off-by: Clement Verna <cverna@tutanota.com>

cverna committed:
This commit makes possible to edit an update using the update title
Fixes #1409

Signed-off-by: Clement Verna <cverna@tutanota.com>

bowlofeggs commented on bodhi/tests/client/test___init__.py:
```diff
@@ -449,6 +449,68 @@ def test_notes_and_notes_file(self):
+                   u'Error: Invalid value for "update": ' \
+                   u'Please provide an Update ID or an Update Title\n'
+
+        self.assertEqual(result.output, expected)
```
I think we probably only need one of the two tests above, since they are both asserting that identifiers that don't match the regexes raise an error and the only difference is which wrong identifier we try. That said, I'm not opposed to keeping both tests if you prefer - your choice ☺

bowlofeggs commented on bodhi/client/__init__.py:
```diff
@@ -175,9 +176,21 @@ def new(user, password, url, **kwargs):
+@click.argument('update', callback=_validate_edit_update)
 @url_option
 def edit(user, password, url, **kwargs):
     """
```
You could mention here that ```UPDATE``` can be the update alias (FEDORA-blah) or the title. I think that would cause ```--help``` to display that info to the user. Optional.

bowlofeggs reviewed (approved):
Looks great! Just a couple of suggestions - you can merge as-is, or you can make the suggested changes and merge (no need for re-review). Nice!

cverna commented:
I have added the doc string to mention than update can be an update id or title. I have also added the change to 2.7.0 release notes.

And thanks for the review .
